src/main/java/org/springframework/data/redis/stream/DefaultStreamMessageListenerContainer.java

```diff
@@ -160,9 +165,22 @@ public void stop() {
 		synchronized (lifecycleMonitor) {
 
 			if (this.running) {
-
-				subscriptions.forEach(Cancelable::cancel);
-
+				subscriptions.stream()
```

Comment:
Running a busy-spin cannot be interrupted. Also, keeping common ForkJoin pool threads busy asks for trouble.

Instead, our Tasks should report a completion future that we can listen to.